test/services/AuditServiceSpec.scala

```diff
@@ -89,22 +108,64 @@ class AuditServiceSpec extends AnyWordSpec with Matchers with MockAuditConnector
+          mockRetrieveAuthenticatorDetails(testJourneyId)(Future.successful(Some(testIndividualDetails)))
+          mockRetrieveES20Response(testJourneyId)(Future.successful(None))
+
+
+          val result: Unit = await(TestService.auditJourney(testJourneyId, testSoleTraderJourneyConfigWithCallingService))
+
+          result mustBe a[Unit]
+
+          verifySendExplicitAuditSoleTraders()
+          auditEventCaptor.getValue mustBe testSoleTraderWithCallingServiceAuditEventJson(identifiersMatch = true)
+        }
+        "there is an sa utr and the journey config has disabled business verification" in {
+          mockRetrieveSoleTraderDetails(testJourneyId)(Future.successful(Some(testSoleTraderWithoutBVCheckDetails)))
+          mockRetrieveIdentifiersMatch(testJourneyId)(Future.successful(Some(true)))
+          mockRetrieveAuthenticatorDetails(testJourneyId)(Future.successful(Some(testIndividualDetails)))
+          mockRetrieveES20Response(testJourneyId)(Future.successful(None))
+
+
+          val result: Unit = await(TestService.auditJourney(testJourneyId, testSoleTraderJourneyConfigWithBVCheckDisabled))
+
+          result mustBe a[Unit]
+
+          verifySendExplicitAuditSoleTraders()
+          auditEventCaptor.getValue mustBe testSoleTraderWithoutBVCheckAuditEventJson(identifiersMatch = true)
+        }
+        "there is an sautr but registration fails" in {
+          mockRetrieveSoleTraderDetails(testJourneyId)(Future.successful(Some(testSoleTraderDetailsRegistrationFailed)))
+          mockRetrieveIdentifiersMatch(testJourneyId)(Future.successful(Some(true)))
+          mockRetrieveAuthenticatorDetails(testJourneyId)(Future.successful(Some(testIndividualDetails)))
+          mockRetrieveES20Response(testJourneyId)(Future.successful(None))
+
+
+          val result: Unit = await(TestService.auditJourney(testJourneyId, testSoleTraderJourneyConfig))
+
+          result mustBe a[Unit]
+
+          verifySendExplicitAuditSoleTraders()
+          auditEventCaptor.getValue mustBe testSoleTraderRegistrationFailedAuditEventJson(identifiersMatch = true)
+        }
         "there is not an sautr" in {
           mockRetrieveSoleTraderDetails(testJourneyId)(Future.successful(Some(testSoleTraderDetailsNoSautr)))
           mockRetrieveIdentifiersMatch(testJourneyId)(Future.successful(Some(true)))
           mockRetrieveAuthenticatorDetails(testJourneyId)(Future.successful(Some(testIndividualDetailsNoSautr)))
           mockRetrieveES20Response(testJourneyId)(Future.successful(None))
 
-          val result: Unit = await(TestService.auditSoleTraderJourney(testJourneyId))
+          val result: Unit = await(TestService.auditJourney(testJourneyId, testSoleTraderJourneyConfig))
 
-          result mustBe()
+          result mustBe a[Unit]
 
           verifySendExplicitAuditSoleTraders()
           auditEventCaptor.getValue mustBe testSoleTraderAuditEventJsonNoSautr(true)
```

Comment:
I'm not a fan of these "auditEventCaptor" verifiers, as we can check exactly what has been sent to the method with normal mockito/scalamock.